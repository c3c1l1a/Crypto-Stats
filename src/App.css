.App {
	background: linear-gradient(
		to right,
		#000,
		#fe5194
	);
}
.back-button-icon {
	fill:  #fff;
	transform: scale(0.5);
}

.home-nav {
	display: flex;
	background: linear-gradient(
      to right,
      #000,
      #fe5194 
    );
}

.show {
	fill: transparent;
}

.hide {
	fill: #fff;
}

