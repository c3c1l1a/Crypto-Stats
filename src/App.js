import { useDispatch } from 'react-redux';
import { useEffect, useState } from 'react';
import { Route, Routes, NavLink } from 'react-router-dom';
import Home from './components/home/Home';
import Details from './components/details/Details';
import { fetchAllCoins } from './redux/coins';

function App() {
  const dispatch = useDispatch();
  const [coinId, setCoinId] = useState('');

  useEffect(() => async () => {
    await dispatch(fetchAllCoins());
  }, []);

  const onClick = () => {
    setCoinId('');
  };

  return (
    <div className="App">
      <NavLink onClick={onClick} to="/">Home</NavLink>
      <Routes>
        <Route exact path="/" element={<Home setCoinId={setCoinId} />} />
        <Route path="details/" element={<Details coinId={coinId} />} />
      </Routes>
    </div>
  );
}

export default App;
