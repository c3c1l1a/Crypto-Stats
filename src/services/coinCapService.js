import axios from 'axios';

const http = axios.create({
  baseURL: 'https://api.coincap.io/v2/assets',
  headers: {
    'Content-type': 'application/json',
  },
});

const getAllCoins = async () => http.get('/', {
  params: {
    limit: 10,
  },
});

const coinCapService = {
  getAllCoins,
};

export default coinCapService;
