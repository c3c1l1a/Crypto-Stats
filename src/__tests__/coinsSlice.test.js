import {
  render, screen, waitFor, fireEvent, act,
} from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import App from '../App';
import store from '../redux/configureStore';
import http from '../services/api';
import coinCapService from '../services/coinCapService';

const coinsSpy = jest.spyOn(coinCapService, 'getAllCoins');
let state = {};

describe('App Test', () => {
  beforeEach(() => {
    state = { 
      data: {
        data: [
          {
            changePercent24Hr: "4.6082010502700002",
            explorer: "https://blockchain.info/",
            id: "bitcoin",
            marketCapUsd: "460942043532.6761288040606192",
            maxSupply: "21000000.0000000000000000",
            name: "Bitcoin",
            priceUsd: "24113.0269735594423344",
            rank: "1",
            supply: "19115893.0000000000000000",
            symbol: "BTC",
            volumeUsd24Hr: "9898331887.6909480283126027",
            vwap24Hr: "23672.2907611691838649",
          }
        ],
    }
      
    };

  });

  afterEach(() => {
    act(() => store.dispatch({
      type: 'coins/FETCH_ALL',
      payload: [],
    }));
  });

  it('Renders all a list coins on load', async () => {
    render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );

    await coinsSpy.mockResolvedValue(state);
    const data = await coinCapService.getAllCoins();
    act(() => store.dispatch({
      type: 'coins/FETCH_ALL/fulfilled',
      payload: data.data,
    }));

    await waitFor(() => {
      expect(coinsSpy).toHaveBeenCalledTimes(1);
      expect(screen.getAllByText('Bitcoin').length).toBeGreaterThan(0);
    });
  });

  it('Displays navigates to item route on click', async () => {
    render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );

    await coinsSpy.mockResolvedValue(state);
    const data = await coinCapService.getAllCoins();
    act(() => store.dispatch({
      type: 'coins/FETCH_ALL/fulfilled',
      payload: data.data,
    }));

    const coinItem = screen.getByText('Bitcoin').closest('a');
    fireEvent.click(coinItem);

    await waitFor(() => {
      expect(window.location.href).toEqual('http://localhost/details/bitcoin');
    });
  });

  it('maintains the snapshots between renders', async () => {
    const tree = render(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );

    await coinsSpy.mockResolvedValue(state);
    const data = await coinCapService.getAllCoins();
    act(() => store.dispatch({
      type: 'coins/FETCH_ALL/fulfilled',
      payload: data.data,
    }));

    await expect(tree).toMatchSnapshot();
  });
});
