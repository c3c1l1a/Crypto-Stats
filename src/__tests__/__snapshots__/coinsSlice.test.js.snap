// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Test maintains the snapshots between renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <a
          class="home-nav"
          href="/"
        >
          <svg
            class="back-button-icon hide"
            height="48"
            width="48"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 44 0 24 20 4l2.8 2.85L5.65 24 22.8 41.15Z"
            />
          </svg>
        </a>
        <div
          class="banner"
        >
          <img
            alt="banner"
            class="banner-image"
            src="banner.png"
          />
          <div
            class="banner-details"
          >
            <h1>
              Crypto Stats
            </h1>
            <p>
              200 Coins
            </p>
          </div>
        </div>
        <div
          class="details"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <a
        class="home-nav"
        href="/"
      >
        <svg
          class="back-button-icon hide"
          height="48"
          width="48"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20 44 0 24 20 4l2.8 2.85L5.65 24 22.8 41.15Z"
          />
        </svg>
      </a>
      <div
        class="banner"
      >
        <img
          alt="banner"
          class="banner-image"
          src="banner.png"
        />
        <div
          class="banner-details"
        >
          <h1>
            Crypto Stats
          </h1>
          <p>
            200 Coins
          </p>
        </div>
      </div>
      <div
        class="details"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
